server/py/services/alerts/crud/alerts.py

```diff
@@ -391,6 +415,27 @@ def reset_alert(self, session: sqlalchemy.orm.Session, project: str, name: str):
                 f"Alert {name} for project {project} does not exist"
             )
 
+        if alert.reset_policy == mlrun.common.schemas.alert.ResetPolicy.MANUAL:
+            # before resetting an alert, update number_of_events if required
+            number_of_events = self._get_number_of_events(alert.id)
+            if number_of_events > alert.criteria.count:
+                alert_state = (
+                    framework.utils.singletons.db.get_db().get_alert_state_dict(
+                        session, alert.id
+                    )
+                )
+                activation_time = alert_state.get("last_updated")
+                activation_id = alert_state.get("full_object", {}).get(
+                    "last_activation_id"
+                )
+                if activation_time and activation_id:
+                    framework.utils.singletons.db.get_db().update_alert_activation_number_of_events(
```

Comment:
Don't we store in the DB also the reset time? We know when it was activated, but we also want to know when it was reset, don't we?

Reply:
@theSaarco no, we don't as it wasn't it the scope originally. We can discuss it and implement as a separate ticket